test/keycloak/irods-http-api-keycloak.Dockerfile

```diff
@@ -1,18 +1,22 @@
-FROM quay.io/keycloak/keycloak:latest as builder
+FROM quay.io/keycloak/keycloak:23.0.6 AS builder
 
-# Configure stuff
-ENV KEYCLOAK_ADMIN=admin
-ENV KEYCLOAK_ADMIN_PASSWORD=admin
-
-# Optimized build thing
+# Build since import command uses --optimized
 RUN /opt/keycloak/bin/kc.sh build
 
+# Use our exported realm made for testing
+COPY example-realm-export.json /realm-export.json
 
-FROM quay.io/keycloak/keycloak:latest
+# Import realm at build time, this will shorten startup time
+RUN /opt/keycloak/bin/kc.sh import --file /realm-export.json
+
+# Use clean image and copy over changes made in builder image
+FROM quay.io/keycloak/keycloak:23.0.6
 COPY --from=builder /opt/keycloak/ /opt/keycloak/
 
-# Use our exported realm
-COPY example-realm-export.json /opt/keycloak/data/import/realm-export.json
+# Configure environment variables
+# TODO: Figure out if this is better left in a compose file...
+ENV KEYCLOAK_ADMIN=admin
+ENV KEYCLOAK_ADMIN_PASSWORD=admin
```

Comment:
If you make these `ARG`s, you can define the `ENV` values based on the `ARG`s and they can be overridden at build time. An example from the development environment to allow for using a different CMake: https://github.com/irods/irods_development_environment/blob/cee974208c809544d75e53ead805b8e8fe850b7a/irods_core_builder.rocky9.Dockerfile#L118-L119

Reply:
Although docker complains about the 'secrets', in both ENV and ARG style, I think it should be fine to leave this as is. It's unlikely that people will define a keycloak admin & password themselves. Having it as an ENV means it is present at runtime, which if I recall, is required.

Reply:
Yes, we should definitely leave them as `ENV` declarations so that they are available at runtime. I was just suggesting adding `ARG`s by which they can be defined / overridden at build time, like this:

```suggestion
# TODO: Figure out if this is better left in a compose file...
ARG admin_username=admin
ENV KEYCLOAK_ADMIN=${admin_username}
ARG admin_password=admin
ENV KEYCLOAK_ADMIN_PASSWORD=${admin_password}
```

Then, someone can override the admin username/password using build args. If it seems overkill, we don't need to do anything here.

Reply:
Didn't even think of that. I think we can leave as is, though. It may be worth reconsidering having the configuration after we implement more automated OIDC tests.

Reply:
Going to leave as is.